pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -9,84 +9,232 @@ import (
+	if kapi.Semantic.DeepEqual(old, cur) {
+		return
+	}
 
-	configController := NewDeploymentConfigController(factory.KubeClient, factory.Client, factory.Codec, recorder)
-
-	return &controller.RetryController{
-		Queue: queue,
-		RetryManager: controller.NewQueueRetryManager(
-			queue,
-			cache.MetaNamespaceKeyFunc,
-			func(obj interface{}, err error, retries controller.Retry) bool {
-				config := obj.(*deployapi.DeploymentConfig)
-				// no retries for a fatal error
-				if _, isFatal := err.(fatalError); isFatal {
-					glog.V(4).Infof("Will not retry fatal error for deploymentConfig %s/%s: %v", config.Namespace, config.Name, err)
-					utilruntime.HandleError(err)
-					return false
-				}
-				// infinite retries for a transient error
-				if _, isTransient := err.(transientError); isTransient {
-					glog.V(4).Infof("Retrying deploymentConfig %s/%s with error: %v", config.Namespace, config.Name, err)
-					return true
-				}
-				utilruntime.HandleError(err)
-				// no retries for anything else
-				if retries.Count > 0 {
-					return false
-				}
-				return true
-			},
-			flowcontrol.NewTokenBucketRateLimiter(1, 10),
-		),
-		Handle: func(obj interface{}) error {
-			config := obj.(*deployapi.DeploymentConfig)
-			copied, err := dcCopy(config)
-			if err != nil {
-				return err
-			}
-			return configController.Handle(copied)
-		},
+	curRC := cur.(*kapi.ReplicationController)
+	glog.V(4).Infof("Replication controller %q updated.", curRC.Name)
+	if dc, err := c.dcStore.GetConfigForController(curRC); err == nil && dc != nil {
+		c.enqueueDeploymentConfig(dc)
 	}
 }
 
+// deleteReplicationController enqueues the deployment that manages a replicationcontroller when
+// the replicationcontroller is deleted. obj could be an *kapi.ReplicationController, or
+// a DeletionFinalStateUnknown marker item.
+func (c *DeploymentConfigController) deleteReplicationController(obj interface{}) {
+	rc, ok := obj.(*kapi.ReplicationController)
+
+	// When a delete is dropped, the relist will notice a pod in the store not
+	// in the list, leading to the insertion of a tombstone object which contains
+	// the deleted key/value.
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			glog.Errorf("Couldn't get object from tombstone %#v", obj)
+			return
+		}
+		rc, ok = tombstone.Obj.(*kapi.ReplicationController)
+		if !ok {
+			glog.Errorf("Tombstone contained object that is not a replication controller %#v", obj)
+			return
+		}
+	}
+	glog.V(4).Infof("Replication controller %q deleted.", rc.Name)
+	if dc, err := c.dcStore.GetConfigForController(rc); err == nil && dc != nil {
+		c.enqueueDeploymentConfig(dc)
+	}
+}
+
+func (c *DeploymentConfigController) enqueueDeploymentConfig(dc *deployapi.DeploymentConfig) {
+	key, err := kcontroller.KeyFunc(dc)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %#v: %v", dc, err)
+		return
+	}
+	c.queue.Add(key)
```

Comment:
> I'm not completely sure of what the controller is trying to do. It seems a little odd that any touch to a DC causes some sort of reconcilation (if I add my own annotation this controller has work to do?). That means that you'll end up self-feeding and adding more and more updates over time, right?

This is the main deployment config controller so it is supposed to reconcile a config on every mutation, as every controller is doing for a resource.